src/transformers/models/parakeet/modular_parakeet.py

```diff
@@ -627,4 +898,106 @@ def generate(
         return sequences
 
 
-__all__ = ["ParakeetForCTC", "ParakeetEncoder", "ParakeetPreTrainedModel"]
+@auto_docstring(
+    custom_intro="""
+    Parakeet TDT model.
+    """
+)
+class ParakeetForTDT(ParakeetPreTrainedModel):
+    config: ParakeetTDTConfig
+
+    def __init__(self, config: ParakeetTDTConfig):
+        super().__init__(config)
+        self.encoder = ParakeetEncoder(config.encoder_config)
+        self.decoder = ParakeetTDTDecoder(config.decoder_config)
+        self.joint = ParakeetTDTJoint(config.joint_config)
+        self.blank_token_id = config.blank_token_id
+        self.max_token_per_frame = 2
+        self.post_init()
+
+    @auto_docstring
+    @can_return_tuple
+    def forward(
+        self,
+        input_features: torch.Tensor,
+        **kwargs: Unpack[TransformersKwargs],
+    ):
+        encoder_outputs = self.encoder(
+            input_features=input_features,
+            **kwargs,
+        )
+
+        logits = self.joint.joint_net(self.joint.enc(encoder_outputs.last_hidden_state)) #[:,:,:self.joint.vocab_size]
+
+        return CausalLMOutput(
+            loss=torch.sum(encoder_outputs.last_hidden_state), # a fake loss here.
+            logits=logits,
+            hidden_states=encoder_outputs.hidden_states,
+            attentions=encoder_outputs.attentions,
+        )
+
+    @torch.no_grad()
+    def generate(
+        self,
+        input_features: torch.Tensor,
+        **kwargs: Unpack[TransformersKwargs],
+    ):
```

Comment:
(Transformers convention) can you imitate `ParakeetForCTC` [here](https://github.com/huggingface/transformers/blob/f8e5ae6a5072cfea78c5d052ebe61b63c9cb10cc/src/transformers/models/parakeet/modular_parakeet.py#L594), namely:
- example in the docstring
- add `return_dict_in_generate`
- `attnetion_mask` not needed for the encoder?